ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveRemoveEmptySingleRules.java

```diff
@@ -192,6 +178,79 @@ public interface JoinRightEmptyRuleConfig extends PruneEmptyRule.Config {
     }
   }
 
+  private static RelNode padWithNulls(RelBuilder builder, RelNode input, RelDataType resultType,
+      boolean leftPadding) {
```

Comment:
nit: you can get rid of the flag argument `leftPadding` by creating two methods each of them responsible only for one type of padding:
```
leftPadWithNulls(RelBuilder builder, RelNode input, RelDataType resultType)
rightPadWithNulls(RelBuilder builder, RelNode input, RelDataType resultType)
```

Reply:
This will lead to duplicating some code that I would like to avoid hence this refactoring.